src/ethereum/prague/vm/eoa_delegation.py

```diff
@@ -144,6 +145,32 @@ def access_delegation(
     return True, address, code, access_gas_cost
 
 
+def access_delegation_read(
```

Comment:
The name of this function doesn't convey that it's reading code. I think it's also in the wrong place.

What do you think of adding this function to every fork, probably in their `environment.py` files. In preceding forks, it'll just return `get_account(evm.env.state, address).code` and in prague it'll have the delegation check.

Reply:
Or maybe add a `get_code` function to `state.py`. It already has the `set_code` function

Reply:
The function is so simple that I just removed it entirely